### PR TITLE
ENH: stats: Use explicit MLE formulas in lognorm.fit and expon.fit

### PR DESCRIPTION
**Pull request by WarrenWeckesser**
The log-normal and exponential distributions have explicit formulas for the maximum likelihood estimation of their parameters, so let's use them when we can.  These formulas are more accurate and *much* faster than using the default numerical optimization routine.

### CODE REVIEW

**Comment by josef-pkt** [comment]
LGTM (I didn't try to remember the exact lognormal shape, scale parameterization and would trust some unit test for no calculation mistake)

File: scipy/misc/doccer.py [hunk 1]
```diff
@@ -149,7 +193,6 @@ def indentcount_lines(lines):
         return 0
     return indentno
 
-
```
Comment (larsoner) [inline scipy/misc/doccer.py]: If you need to make any more changes to this function, might as well revert this to help with PEP8 / #8388. But it's not worth a separate commit unless you in a particularly generous, pedantic, and/or bored mood.

Comment (WarrenWeckesser) [reply scipy/misc/doccer.py]: Whoops, removing that line blank line was indeed a mistake.  I think I was doing and undoing some bulk text movement and must have removed one line too many.  I'll fix that.

**Comment by WarrenWeckesser** [comment]
@larsoner The mistaken removal of the blank line has been reverted.

**Comment by ev-br** [comment]
Thanks Warren

**Pull request merged**